lambdas/backend-api/src/utils/validate.ts

```diff
@@ -0,0 +1,43 @@
+import { ErrorCase } from 'nhs-notify-backend-client';
+import { z } from 'zod';
+import { ApplicationResult, failure, success } from './result';
+
+const formatZodErrors = (error: z.ZodError) => {
+  const formattedErrors: Record<string, string[]> = {};
+
+  for (const issue of error.issues) {
+    const [fieldName] = issue.path;
+    const errorMessage = issue.message;
+
+    if (!formattedErrors[fieldName]) {
+      formattedErrors[fieldName] = [];
+    }
+
+    formattedErrors[fieldName].push(errorMessage);
+  }
+
+  return Object.fromEntries(
+    Object.entries(formattedErrors).map(([key, value]) => [
+      key,
+      value.join(', '),
+    ])
+  );
+};
+
+export const validate = async <T extends z.Schema>(
+  $schema: T,
+  dto: unknown
+): Promise<ApplicationResult<z.infer<T>>> => {
+  const { error, data } = await $schema.safeParseAsync(dto);
```

Comment:
So, I thought I had issues with some async issues in Zod due to the regex refine. So I used the `safeParseAsync` function to see if this helps. It didn't see https://github.com/NHSDigital/nhs-notify-web-template-management/pull/197/files#r1867773165.

Now I don't want to change it back (too many changes).